_books/ion-1-1/src/security-considerations.md

```diff
@@ -0,0 +1,156 @@
+nest an arbitrary number of loops to create a macro expansion with a polynomial growth rate. Using the 
+[`repeat`](macros/system_macros.md#repeat) macro, we can create classes of inputs with expansions that grow 
+exponentially in relation to the input.
+
+For example, this input is less than 250 characters when encoded as Ion text (and omitting all optional whitespace). 
+In Ion binary, it requires only 74 bytes. For each additional level of nesting, only 20 additional characters (text) or
+6 additional bytes (binary) are required, but it increases the number of expanded values by 2147483647 times.
+```ion
+$ion_1_1
+(:repeat 2147483647
+  (:repeat 2147483647
+    (:repeat 2147483647
+      (:repeat 2147483647
+        (:repeat 2147483647
+          (:repeat 2147483647
+            (:repeat 2147483647
+              (:repeat 2147483647
+                (:repeat 2147483647
+                  (:repeat 2147483647
+                    (:repeat 2147483647 "abc")))))))))))
+```
+The expansion of these e-expressions results in a stream of ~450 [googol](https://en.wikipedia.org/wiki/Googol) string
+values. Any attempt to hold all of this in memory or write it to disk will exhaust all available resources and 
+eventually fail. Even an attempt to count the length of the stream, while it may theoretically succeed if using an 
+appropriate `BigInteger` type, will require a considerable amount of CPU operations (over a googol), and even the 
+fastest processors will require many millennia to completely count the number of values in the stream.
+
+Even without using `repeat` or `for`, a [Billion laughs attack](https://en.wikipedia.org/wiki/Billion_laughs_attack)
+could exist for any data format with macro expansion, and it is certainly possible with Ion 1.1.
+```ion
+$ion_1_1
+(:add_macros (macro lol0 () "lol")
+             (macro lol1 () (.values (.lol0) (.lol0) (.lol0) (.lol0) (.lol0) (.lol0) (.lol0) (.lol0) (.lol0) (.lol0)))
+             (macro lol2 () (.values (.lol1) (.lol1) (.lol1) (.lol1) (.lol1) (.lol1) (.lol1) (.lol1) (.lol1) (.lol1)))
+             (macro lol3 () (.values (.lol2) (.lol2) (.lol2) (.lol2) (.lol2) (.lol2) (.lol2) (.lol2) (.lol2) (.lol2)))
+             (macro lol4 () (.values (.lol3) (.lol3) (.lol3) (.lol3) (.lol3) (.lol3) (.lol3) (.lol3) (.lol3) (.lol3)))
+             (macro lol5 () (.values (.lol4) (.lol4) (.lol4) (.lol4) (.lol4) (.lol4) (.lol4) (.lol4) (.lol4) (.lol4)))
+             (macro lol6 () (.values (.lol5) (.lol5) (.lol5) (.lol5) (.lol5) (.lol5) (.lol5) (.lol5) (.lol5) (.lol5)))
+             (macro lol7 () (.values (.lol6) (.lol6) (.lol6) (.lol6) (.lol6) (.lol6) (.lol6) (.lol6) (.lol6) (.lol6)))
+             (macro lol8 () (.values (.lol7) (.lol7) (.lol7) (.lol7) (.lol7) (.lol7) (.lol7) (.lol7) (.lol7) (.lol7)))
+             (macro lol9 () (.values (.lol8) (.lol8) (.lol8) (.lol8) (.lol8) (.lol8) (.lol8) (.lol8) (.lol8) (.lol8)))
+             (macro lolz () (.lol9)) )
+(:lolz)
```

Comment:
We should make a `TODO` to track adding these examples to `ion-tests`.

Reply:
I've added it to https://github.com/amazon-ion/ion-tests/issues/88